libs/blocks/modal/modal.js

```diff
@@ -52,12 +52,20 @@ function getCustomModal(custom, dialog) {
   dialog.append(custom.content);
 }
 
+async function decorateModal(dialog) {
```

Comment:
Ideally we don't have "non-modal" code in modal.js.  Not sure what an alternative would be (yet)

Reply:
agree,  and am not sure either. 
i'll put my thinking cap on today though.